runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java

```diff
@@ -420,22 +334,22 @@ public void initChannel(SocketChannel ch) throws Exception {
         }
     }
 
-    synchronized void connectChannel(Bootstrap b, long c) {
+    synchronized void connectChannel(Bootstrap b, long c)
+        throws ConnectException, NoRouteToHostException, SocketException
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.net.ConnectException' which is a subclass of 'java.net.SocketException'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.net.NoRouteToHostException' which is a subclass of 'java.net.SocketException'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.net.SocketException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)